spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,397 @@
+require "spec"
+require "yaml"
+
+private class ObjectStub
+  def self.new(pp : YAML::PullParser)
+    pp.read_scalar.should eq "hello"
+    new
+  end
+end
+
+describe "Object.from_yaml" do
+  it "should create call .new on with a YAML::PullParser" do
+    ObjectStub.from_yaml("hello")
+  end
+end
+
+describe "Nil.from_yaml" do
+  values = ["", "NULL", "Null", "null", "~"]
+  values.each do |value|
+    it "should serialize and deserialize" do
+      Nil.from_yaml(nil.to_yaml).should eq nil
+    end
+
+    it "should return nil if #{value}" do
+      # Test will an array since a standalone empty value shows as STREAM_END
+      Array(Nil).from_yaml("- " + value).should eq [nil]
+    end
+
+    it "should raise if #{value} is quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "\"#{value}\""
+      end
+    end
+
+    it "should raise if #{value} is quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "'#{value}'"
+      end
+    end
+  end
+
+  it "should raise if not a null value" do
+    expect_raises(YAML::ParseException) do
+      Nil.from_yaml "hello"
+    end
+  end
+end
+
+describe "Bool.from_yaml" do
+
+  describe "true values" do
+    it "should serialize and deserialize" do
+      Bool.from_yaml(true.to_yaml).should eq true
+    end
+
+    values = ["true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"]
```

Comment:
I don't like adding so many specs for each single plain value. It would probably be better to put all variations together in one spec to have three specs with inner loops of 11 values instead of 33 individual specs which test exactly the same behaviour. This just adds dozens of mostly meaningless specs. I think the different keywords can be safely combined into a single spec.
This applies to the other ocations as well.